### ISSUE TITLE
MDAO and its indexes do not support unsubscribing from updates

### ISSUE THREAD

**Issue by mdittmer** [foam-framework/foam2#633]
`mdao.select(SomeSink({ put: (o, sub) => sub.detach() }))` will fail because `sub` is `undefined`. In addition to "passing something" here, the MDAO needs to pass around enough state to honour the `detach()`.
